modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v4_2_1/test/WorkflowContextUpgradeProcessTest.java

```diff
@@ -0,0 +1,182 @@
+			RandomTestUtil.randomString(),
+			ServiceContextTestUtil.getServiceContext(group.getGroupId()));
+
+		String workflowContext = null;
+
+		try (Connection connection = DataAccess.getConnection();
+			PreparedStatement preparedStatement1 = connection.prepareStatement(
+				StringBundler.concat(
+					"select kaleoLogId, workflowContext from KaleoLog where ",
+					"workflowContext is not null and workflowContext like ",
+					"'%ServiceContext%'"));
+
+			ResultSet resultSet = preparedStatement1.executeQuery()) {
+
+			while (resultSet.next()) {
+				workflowContext = resultSet.getString("workflowContext");
+			}
+
+			JSONObject workflowContextJSONObject = null;
+
+			if (workflowContext != null) {
+				workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
+					workflowContext);
+			}
+			else {
+				throw new IllegalStateException("workflowContext is null");
+			}
+
+			JSONObject jsonObject1 = workflowContextJSONObject.getJSONObject(
+				"map");
+
+			JSONObject jsonObject2 = jsonObject1.getJSONObject(
+				"serviceContext");
+
+			jsonObject2.remove("javaClass");
+
+			jsonObject2.put(
+				"javaClass",
+				"com.liferay.headless.common.spi.service.context." +
+					"ServiceContextUtil$1");
+
+			try (PreparedStatement preparedStatement2 =
+					connection.prepareStatement(
+						StringBundler.concat(
+							"update KaleoLog set workflowContext = ? where ",
+							"workflowContext is not null and workflowContext ",
+							"like '%com.liferay.portal ",
+							".kernel.service.ServiceContext%'"))) {
+
+				preparedStatement2.setString(
+					1, workflowContextJSONObject.toString());
+
+				preparedStatement2.executeUpdate();
+
+				connection.close();
+			}
+		}
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME, LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+
+		String workflowContext = null;
+
+		try (Connection connection = DataAccess.getConnection()) {
+			PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select workflowContext from KaleoLog where workflowContext " +
+					"is not null");
+
+			try (ResultSet resultSet = preparedStatement1.executeQuery()) {
+				while (resultSet.next()) {
+					workflowContext = resultSet.getString("workflowContext");
+				}
+
+				connection.close();
+			}
+		}
+
+		JSONObject workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
+			workflowContext);
+
+		JSONObject jsonObject1 = workflowContextJSONObject.getJSONObject("map");
+
+		JSONObject jsonObject2 = jsonObject1.getJSONObject("serviceContext");
+
+		Assert.assertNotEquals(
+			"com.liferay.headless.common.spi.service.context." +
+				"ServiceContextUtil$1",
+			jsonObject2.get("javaClass"));
```

Comment:
@JhossephG please simplify by using the local service to `get` and `assert`:

Something like:

```java
KaleoInstance kaleoInstance = _kaleoInstanceLocalService.fetchKaleoInstance(_kaleoInstance.getKaleoInstanceId());

JSONObject workflowContextJSONObject = JSONFactoryUtil.createJSONObject(
	kaleoInstance.getWorkflowContext());

...
```